solutions/09/script.js

```diff
@@ -1,30 +1,63 @@
-function validateCardNumber(number) {
-    var regex = new RegExp("^[0-9]{16}$");
-    if (!regex.test(number))
-        return false;
+function validateCardNumber(value) {
+    // Accept only digits, dashes or spaces
+	if (/[^0-9-\s]+/.test(value)) return false;
 
-    return luhnCheck(number);
+	// The Luhn Algorithm. It's so pretty.
+	let nCheck = 0, bEven = false;
+	value = value.replace(/\D/g, "");
+
+	for (let n = value.length - 1; n >= 0; n--) {
+		let cDigit = value.charAt(n),
+			  nDigit = parseInt(cDigit, 10);
+
+		if (bEven && (nDigit *= 2) > 9) nDigit -= 9;
+
+		nCheck += nDigit;
+		bEven = !bEven;
+	}
+	return (nCheck % 10) == 0;
 }
 
-function luhnCheck(val) {
-    var sum = 0;
-    for (var i = 0; i < val.length; i++) {
-        var intVal = parseInt(val.substr(i, 1));
-        if (i % 2 == 0) {
-            intVal *= 2;
-            if (intVal > 9) {
-                intVal = 1 + (intVal % 10);
-            }
+function whatCard(value) {
+    const icon = document.querySelector('.icon');
+    switch(value[0]) {
+        case '3': {
+            icon.classList.add('fa-cc-amex');
+            break;
+        }
+        case '4': {
+            icon.classList.add('fa-cc-visa');
+            break;
+        }
+        case '5': {
+            icon.classList.add('fa-cc-mastercard');
+            break;
+        }
+        case '6': {
+            icon.classList.add('fa-cc-discover');
+            break;
+        }
+        default: {
+            icon.className = "";
+            icon.classList.add('icon');
+            icon.classList.add('fa');
         }
-        sum += intVal;
     }
-    return (sum % 10) == 0;
 }
-const myCard = document.querySelector('#card-number-field');
 
-function validate() {
-    console.log("validate -> myCard.value", myCard.value)
-    validateCardNumber(myCard.value);
+function formValidation(e)
+{
+    e.preventDefault();
+    const myCard = document.querySelector('#cardNumber');
+    const val = myCard.value;
```

Comment:
The same, you don't need to get buttons every time when you click "validate form"

Reply:
Done :)